Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java

```diff
@@ -139,6 +148,40 @@ protected void addCompletions(@NotNull final CompletionParameters parameters,
            });
   }
 
+  private static void appendRuntimeCompletion(@NotNull final CompletionParameters parameters,
+                                              CompletionResultSet resultSet) {
+    final PsiFile originalFile = parameters.getOriginalFile();
+    final Project project = originalFile.getProject();
+    final PsiElement contextElement = originalFile.getContext();
+    if (contextElement == null) {
+      return;
+    }
+
+    final VirtualFile contextFile = contextElement.getContainingFile().getVirtualFile();
+    final int contextOffset = contextElement.getTextOffset();
+
+    final Document dummyDocument = FileDocumentManager.getInstance().getDocument(originalFile.getVirtualFile());
+    if (dummyDocument == null) {
+      return;
+    }
+
+    final String code = dummyDocument.getText();
+    final int codeOffset = parameters.getOffset();
+
+    final DartAnalysisServerService das = DartAnalysisServerService.getInstance(project);
+    final RuntimeCompletionResult completionResult =
+      das.execution_getSuggestions(originalFile.getVirtualFile(),
+                                   code, codeOffset,
+                                   contextFile, contextOffset,
+                                   Lists.newArrayList(), Lists.newArrayList());
+    if (completionResult.suggestions != null) {
```

Comment:
I've got NPE here

Reply:
You probably have a bit older SDK, which does not implement this request yet.
So, we get a timeout and return null.
Of course this is still a bug in this PR - we need to check for null.

I will mark the method nullable as well.